src/wallet/rpcwallet.cpp

```diff
@@ -2554,6 +2556,230 @@ UniValue fundrawtransaction(const UniValue& params, bool fHelp)
+        throw JSONRPCError(RPC_MISC_ERROR, strError);
+    }
+
+    CAmount nDelta = nNewFee - nOldFee;
+
+    {
+        LOCK(mempool.cs);
+        auto it = mempool.mapTx.find(hash);
+        if (it != mempool.mapTx.end()) {
+
+            // Tx with descendants
+            // --------------------
+            // Idea:  When a tx is bumped, its descendants (if any) are evicted from the mempool.
+            // Policy is that when you replace a tx, the total fees in the mempool cannot go down.
+            // So when you bump a tx that has children, you have to bump your fee by the sum of all of the
+            // descendants, plus the new bumped tx's relay fee.
+            //
+            // If the bumped fee is less than what's required because of children, we fail.  This is a different
+            // situation than bumping the fee to pay the minimum relay fee. Here, we can't be sure whether the user
+            // really wants to pay full price for all of the child transactions.  If so, the user can set payTxFee
+            // and run the command again.
+            //
+            CAmount nFeesWithDescendants = it->GetModFeesWithDescendants();
+            if (nNewFee < nFeesWithDescendants + ::minRelayTxFee.GetFee(txSize)) {
+                std::string strError = strprintf("Insufficent fee due to child transactions, the bumped fee must be at least: %s", FormatMoney(nFeesWithDescendants + ::minRelayTxFee.GetFee(txSize)));
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strError);
+            }
+        }
+        else
+        {
+            // Tx not in mempool
+            // -----------------
+            // If the tx is in the wallet but not the mempool, we cannot check its descendants' fees.
+            //
+            // Not being in the mempool likely means that it was evicted for having a low fee.
+            // In this situation, we may not need RBF for this node (if the tx is not in the mempool,
+            // then it doesn't need to be replaced), but it may still be in the mempol of peers
+            // (perhaps a peer has allocated more space for the mempool).
+            //
+            // Our approach is to go ahead and bump/commit/relay the transaction.  In the event that
+            // the tx does have children and the fee is insufficient to cover, the peer(s) will reject
+            // the tx on that basis, so we warn the user of this possibility.
+            //
+            LogPrint("rpc", "Warning: bumping fee on tx that is not in the mempool; if it has child transactions, it may be rejected by peers\n");
+        }
+    }
+
+    // Now modify the output to increase the fee.
+    // Output must be able to pay the increased fee, without being reduced to dust
+    CMutableTransaction tx(wtx);
+    CTxOut* poutput = &(tx.vout[nOutput]);
+    if (poutput->nValue >= nDelta + poutput->GetDustThreshold(::minRelayTxFee)) {
```

Comment:
I checked earlier that the bumped fee pays for the new relay fee;  I'm not following as to the relationship between nDelta and dust.